code/lib/core-server/src/utils/get-new-story-file.test.ts

```diff
@@ -0,0 +1,83 @@
+import { describe, expect, it, vi } from 'vitest';
+import { getNewStoryFile } from './get-new-story-file';
+import path from 'path';
+
+vi.mock('@storybook/core-common', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('@storybook/core-common')>();
+  return {
+    ...actual,
+    getProjectRoot: vi.fn().mockReturnValue(require('path').join(__dirname)),
+  };
+});
+
+describe('get-new-story-file', () => {
+  it('should create a new story file (TypeScript)', async () => {
+    const { exportedStoryName, storyFileContent, storyFilePath } = await getNewStoryFile(
+      {
+        filepath: 'src/components/Page.tsx',
+        componentExportName: 'Page',
+        default: false,
+      },
+      {
+        presets: {
+          apply: (val: string) => {
+            if (val === 'framework') {
+              return Promise.resolve('@storybook/nextjs');
+            }
+          },
+        },
+      } as any
+    );
+
+    expect(exportedStoryName).toBe('Default');
+    expect(storyFileContent).toMatchInlineSnapshot(`
+      "import type { Meta, StoryObj } from '@storybook/nextjs';
+
+      import { Page } from './Page';
+
+      const meta = {
+        component: Page
+      } satisfies Meta<typeof Page>
+
+      export default meta;
+
+      type Story = StoryObj<typeof meta>;
+
+      export const Default: Story = {}"
+    `);
+    expect(storyFilePath).toBe(path.join(__dirname, 'src', 'components', 'Page.stories.tsx'));
+  });
+
+  it('should create a new story file (JavaScript)', async () => {
+    const { exportedStoryName, storyFileContent, storyFilePath } = await getNewStoryFile(
+      {
+        filepath: 'src/components/Page.jsx',
+        componentExportName: 'Page',
+        default: true,
+      },
+      {
+        presets: {
+          apply: (val: string) => {
+            if (val === 'framework') {
+              return Promise.resolve('@storybook/nextjs');
+            }
+          },
+        },
+      } as any
+    );
+
+    expect(exportedStoryName).toBe('Default');
+    expect(storyFileContent).toMatchInlineSnapshot(`
+      "import Component from './Page';
```

Comment:
`componentExportName` for a default export would be `default`. And since `default` is a reserved keyword, I would suggest using a static string instead. It doesn't have to be `Component`, though.